src/main/kotlin/org/rust/toml/CargoTomlDependenciesCompletionProvider.kt

```diff
@@ -0,0 +1,136 @@
+                    val table = key.ancestorStrict<TomlTable>() ?: return@withInsertHandler
+                    if (table.entries.isEmpty()) {
+                        context.document.insertString(
+                            context.selectionEndOffset + 1,
+                            "\nversion = \"${variant.maxVersion}\""
+                        )
+                    }
+                }
+        })
+    }
+}
+
+/** @see CargoTomlPsiPattern.inSpecificDependencyKeyValue */
+class CargoTomlSpecificDependencyVersionCompletionProvider : AbstractTomlKeyValueCompletionProvider() {
+    override fun completeDependencyKey(key: TomlKey, result: CompletionResultSet) {
+        val keyValue = key.parent as? TomlKeyValue
+            ?: error("PsiElementPattern must not allow keys outside of TomlKeyValues")
+        val dependencyNameKey = getDependencyKeyFromTableHeader(keyValue)
+
+        val version = getCrateLastVersion(dependencyNameKey) ?: return
+        result.addElement(LookupElementBuilder.create("version = \"$version\""))
+    }
+
+    override fun completeDependencyValue(keyValue: TomlKeyValue, result: CompletionResultSet) {
+        val dependencyNameKey = getDependencyKeyFromTableHeader(keyValue)
+        val version = getCrateLastVersion(dependencyNameKey) ?: return
+
+        result.addElement(
+            LookupElementBuilder.create(version)
+                .withInsertHandler(StringValueInsertionHandler(keyValue))
+        )
+    }
+
+    private fun getDependencyKeyFromTableHeader(keyValue: TomlKeyValue): TomlKey {
+        val table = keyValue.parent as? TomlTable
+            ?: error("PsiElementPattern must not allow keys outside of TomlTable")
+        return table.header.names.lastOrNull()
+            ?: error("PsiElementPattern must not allow KeyValues in tables without header")
+    }
+}
+
+/** Inserts `=` between key and value if missed and wraps inserted string with quotes if needed */
+private class StringValueInsertionHandler(val keyValue: TomlKeyValue) : InsertHandler<LookupElement> {
+    override fun handleInsert(context: InsertionContext, item: LookupElement) {
+        var startOffset = context.startOffset
+        val value = context.getItemOfType<TomlValue>()
+        val hasEq = keyValue.children.find { it.elementType == TomlElementTypes.EQ } != null
+        val hasQuotes = value != null && (value !is TomlLiteral || value.literalType != TomlElementTypes.NUMBER)
+
+        if (!hasEq) {
+            context.document.insertString(startOffset - if (hasQuotes) 1 else 0, "= ")
+            PsiDocumentManager.getInstance(context.project).commitDocument(context.document)
+            startOffset += 2
+        }
+
+        if (!hasQuotes) {
+            context.document.insertString(startOffset, "\"")
+            context.document.insertString(context.selectionEndOffset, "\"")
+        }
+    }
+}
+
+private inline fun <reified T : PsiElement> InsertionContext.getItemOfType(strict: Boolean = false): T? =
```

Comment:
Let's drop this function and use `org.rust.lang.core.completion.LookupElementsKt#getElementOfType` instead